### PR TITLE
Use `endterm` attribute when present

### LINKED ISSUE #6619: endterm attribute of xref element of DocBook is not respected
Take this `input.db` file:

```
<chapter xml:id="_responsibility">
<simpara>See regular link: <xref linkend="SEC-Definitions"/>.</simpara>
<simpara>See link with endterm: <xref linkend="SEC-Definitions" endterm="link"/>.</simpara>
</chapter>
<chapter xml:id="SEC-Definitions">
<title>Definitions</title>
<simpara>Some definitions.</simpara>
</chapter>
```

Convert it to `output.docx`:

    pandoc.exe -s input.db -o output.docx

See the generated output:

image

*Expected:* The second link has the text "link". 

*Actual:* The second link has the text "???".

[The docs say](https://tdg.docbook.org/tdg/5.1/xref.html):

> If the endterm attribute is specified on xref, the content of the element pointed to by endterm will be used as the text of the cross-reference.

### PR DESCRIPTION
**Pull request by argent0**
Fixes #6619

I should have also updated the tests at:
```
test/docbook-xref.docbook
test/docbook-xref.native
```

But, how do I run "just" those tests?
What should be the output in pandoc's representation?

### CODE REVIEW

**Comment by jgm** [comment]
Thank you!

To run particular tests (pattern-matching on their names), use
the `-p` option:

    cabal install pandoc --enable-tests
    cabal test --test-options='-p markdown'

Or with stack:

    stack test --test-arguments='-p markdown'

The easiest way to update these golden tests is to update the source docbook-xref.docbook, then run the tests with `--accept` as one of the test arguments/options.  This will update docbook-xref.native.  Of course at this point it's very important to inspect the diff carefully to make sure the new output is correct.

**Comment by argent0** [comment]
@jgm Thank you for maintaining `pandoc` and for the help regarding testing.

File: test/docbook-xref.native [hunk 1]
```diff
@@ -4,7 +4,7 @@ Pandoc (Meta {unMeta = fromList [("title",MetaInlines [Str "An",Space,Str "Examp
 ,BulletList
  [[Para [Str "A",Space,Str "straight",Space,Str "link",Space,Str "generates",Space,Str "the",SoftBreak,Str "cross-reference",Space,Str "text:",Space,Link ("",[],[]) [Str "The",Space,Str "Second",Space,Str "Chapter"] ("#ch02",""),Str "."]]
  ,[Para [Str "A",Space,Str "link",Space,Str "to",Space,Str "an",Space,Str "element",Space,Str "with",Space,Str "an",SoftBreak,Str "XRefLabel:",SoftBreak,Link ("",[],[]) [Str "Chapter",Space,Str "the",Space,Str "Third"] ("#ch03",""),Str "."]]
- ,[Para [Str "A",Space,Str "link",Space,Str "with",Space,Str "an",SoftBreak,Str "EndTerm:",SoftBreak,Link ("",[],[]) [Str "Chapter",Space,Str "4"] ("#ch04",""),Str "."]]
+ ,[Para [Str "A",Space,Str "link",Space,Str "with",Space,Str "an",SoftBreak,Str "EndTerm:",SoftBreak,Link ("",[],[]) [Str "ch04short"] ("#ch04",""),Str "."]]
```
Comment (jgm) [inline test/docbook-xref.native:7]: This doesn't look right, does it?

Comment (jgm) [reply test/docbook-xref.native:7]: Also, maybe add a test that tests explicitly for the case being fixed?

Comment (argent0) [reply test/docbook-xref.native:7]: Yes, after carefully reading the [specification](https://tdg.docbook.org/tdg/5.1/xref.html) I find that #6619 isn't a bug. And pandoc is doing as the specification says.

In particular:

```
<simpara>See link with endterm: <xref linkend="SEC-Definitions" endterm="link"/>.</simpara>
```

`link` should be the id of the element whose content should be used for the link text.

> If the endterm attribute is specified on xref, the content of the element pointed to by endterm will be used as the text of the cross-reference.

So this pull request can be discarded.

**Comment by jgm** [comment]
I'm sorry I didn't notice that it wasn't a real bug before tagging it "good first issue." But thanks for working on it!

**Pull request closed without merging**